src/main/java/io/logz/jmx2logzio/listener/prometheus/TimeSeriesDiskQueue.kt

```diff
@@ -0,0 +1,82 @@
+package io.logz.jmx2logzio.listener.prometheus
+
+import io.logz.sender.SenderStatusReporter
+import io.logz.sender.com.bluejeans.common.bigqueue.BigQueue
+import io.logz.sender.exceptions.LogzioParameterErrorException
+import java.io.Closeable
+import java.io.File
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+
+class TimeSeriesDiskQueue(
```

Comment:
there is nothing in this class which is TimeSeries related - couldn't you use the original class?

Reply:
It has a private constructor, and only has public builder, but the builder requires a LogzioSender instance,
to allow writing build flow like
LogzioSender.bulder()
  .withDiskQueue()
    .withDir(...)
  .endDiskQueue
  .withAnotherLogzioSenderParam()

when a builder builds a builder that builds the builder of the builder, it's a build shitstorm
and a code you can't reuse

so I had to copy the diskQueue 🤦